hbase/src/test/java/org/apache/metamodel/hbase/DeleteRowTest.java

```diff
@@ -0,0 +1,181 @@
+            fail("Should get an exception that rowKey is null");
+        } catch (IllegalArgumentException e) {
+            assertEquals("Can't delete a row without having tableName or rowKey", e.getMessage());
+        }
+    }
+
+    /**
+     * Goodflow. Deleting a row, that doesn't exist, should not throw an exception
+     */
+    @Test
+    public void testDeletingNotExistingRow() {
+        try {
+            final HBaseTable existingTable = createAndAddTableToDatastore(TABLE_NAME, HBaseDataContext.FIELD_ID, CF_FOO,
+                    CF_BAR);
+
+            checkRows(false, false);
+            final HBaseRowDeletionBuilder rowDeletionBuilder = (HBaseRowDeletionBuilder) getUpdateCallback().deleteFrom(
+                    existingTable);
+            rowDeletionBuilder.setKey(RK_1);
+            rowDeletionBuilder.execute();
+            checkRows(false, false);
+        } catch (Exception e) {
+            fail("Should not get an exception that the row doesn't exist.");
+        }
+    }
+
+    /**
+     * Goodflow. Deleting a row, which has an empty rowKey value, should not throw an exception
+     */
+    @Test
+    public void testUsingAnEmptyRowKeyValue() {
+        try {
+            final HBaseTable existingTable = createAndAddTableToDatastore(TABLE_NAME, HBaseDataContext.FIELD_ID, CF_FOO,
+                    CF_BAR);
+
+            checkRows(false, false);
+            final HBaseRowDeletionBuilder rowDeletionBuilder = (HBaseRowDeletionBuilder) getUpdateCallback().deleteFrom(
+                    existingTable);
+            rowDeletionBuilder.setKey("");
+            rowDeletionBuilder.execute();
+            checkRows(false, false);
+        } catch (Exception e) {
+            fail("Should not get an exception that the rowkey is empty.");
+        }
+    }
+
+    /**
+     * Goodflow. Deleting a row succesfully.
+     */
+    @Test
+    public void testDeleteRowSuccesfully() {
+        try {
+            final HBaseTable existingTable = createAndAddTableToDatastore(TABLE_NAME, HBaseDataContext.FIELD_ID, CF_FOO,
+                    CF_BAR);
+            final LinkedHashMap<HBaseColumn, Object> row = createRow(existingTable, HBaseDataContext.FIELD_ID, CF_FOO,
+                    CF_BAR, false);
+            final List<HBaseColumn> columns = getHBaseColumnsFromRow(row);
+
+            checkRows(false, false);
+            final HBaseRowInsertionBuilder rowInsertionBuilder = getUpdateCallback().insertInto(existingTable, columns);
+            setValuesInInsertionBuilder(row, rowInsertionBuilder);
+            rowInsertionBuilder.execute();
+            checkRows(true, false);
+            final HBaseRowDeletionBuilder rowDeletionBuilder = (HBaseRowDeletionBuilder) getUpdateCallback().deleteFrom(
+                    existingTable);
+            rowDeletionBuilder.setKey(RK_1);
+            rowDeletionBuilder.execute();
+            checkRows(false, false);
+        } catch (Exception e) {
```

Comment:
Same like earlier, don't catch the exception but have the method throw `IOException` instead.